### ISSUE TITLE
🔃 🔗 🚀 Feature Request: Show/Hide Elements (widgets & Capabilities) per User or CPT AKA role manager

### ISSUE THREAD

**Issue by mizejewski** [elementor/elementor#1314]
## Description

I hope you’ll consider adding finer controls for Elementor. We’re using Genesis and the Dynamik child theme, and they both offer permissions that I can see/control as an admin and set for each user. See attached 01 and 02. 

<img width="669" alt="01-users-gen" src="https://cloud.githubusercontent.com/assets/974442/23088159/be012f46-f52e-11e6-92c2-5c3412eacd94.png">

<img width="698" alt="02-users-dyn" src="https://cloud.githubusercontent.com/assets/974442/23088160/be01ba24-f52e-11e6-938a-8b2f06fc6a15.png">

The suggested method for adding complex content to toggles/accordions/tabs is to use *embed template* (see https://github.com/pojome/elementor/issues/303) via Ariel Klikstein. However, it appears that only admins have access to edit items in the Elementor Library

**Comment by ehausen** [comment]
Yes some more fine grained control is needed.

For example let Editors edit texts and such, but not drag around elements or change styles like paddings and color.

**Comment by arnii5** [comment]
Interested as well in having this option where you can let a client only edit the text and images (content) but not accidentally change the structure / design.

**Comment by arielk** [comment]
More references for this issue: #681, #653, #481,#536,

**Comment by atMaarten** [comment]
Great idea indeed! 

It would also be nice to hide all the WordPress widgets, since most of them are duplicates of elementor elements.

**Comment by sayantandas** [comment]
Hi,
Is this idea being worked upon?

**Comment by wellcommunications** [comment]
What is the status of this feature request?

**Comment by chi-soul** [comment]
After comparing Elementor to Divi Builder it arms this option / feature Is the only thing that they offer that Elementor does not.   Is there any update on the possibility of adding this feature?

Amy help you can offer would be greatly appreciated. Thanks in advance. 


Lock

Locking is a function that is only available via Divi’s right click controls. Once an item has been locked, it cannot be moved, edited or deleted until it has been unlocked. Only users with the Administrator role have the ability to lock and unlock items. This means you can assign your clients a lesser role, such as Editor, and they won’t be able to edit any item that has been locked. This is a great way to control what your clients can and cannot edit. Often times you might want to lock everything except for text modules, for example, which would allow your client to edit their content, but would not allow them to adjust the settings of more advanced modules.
[https://drive.google.com/file/d/1rCuEFsSjq4stoCSM3ay5qd7J_95sEw78/view?usp=drivesdk](url)

**Comment by auban** [comment]
This feature would be really appreciated

**Comment by sharonrasta** [comment]
will appreciate any update for this request

**Comment by florianlg4** [comment]
It would be great to have this feature! So our clients can edit the content without editing and breaking the layout. It will be very appreciate!

**Comment by jrc2** [comment]
This would be great to keep people from messing around with the layout!

**Comment by srcek** [comment]
The issue i have with new Role manager is that while it block non-admin users from messing with layout while inside Elementor editor it still allows a user to click that "Back to WordPress Editor" button and essentially override the Elementor template. The same goes for other way around. User clicks blue "Edit with Elementor" button on page or post which doesn't require Elementor interface and overrides everything he or she created in standard WordPress editor. Switching between those two should be lockable to avoid confusion and unnecessary maintenance calls.

**Comment by jrc2** [comment]
@srcek Sorry, where is this new role manager?

**Comment by srcek** [comment]
@jrc2 You can find it in your admin panel Elementor -> Role Manager. 
More info here: https://elementor.com/role-manager/

**Comment by jrc2** [comment]
@srcek Ah, I see, it's pro only. I'll be getting that later most likely, but it's not needed for this project. Thanks!

**Comment by gtermini** [comment]
Visual Composer and WP Bakery Page Builder have a much more granular role management feature, where you can not only decide what elements each role can edit, but also what elements each role can use when creating a new page. For an enterprise content management platform, I will have multiple roles (editors, simple authors, etc), and I might want to give the editors access to some of the fancy elements (accordions, maps, etc) but prevent authors from using those elements at all. Any plans to implement a more granular role/element control?

**Comment by Pavandevjoshi** [comment]
Hide Widgets without Elementor Pro, I need help to hide All widgets except Text Editor for user access.
We are using Elementor Free and wants to have only one functionality of Pro, i.e. Access to edit content only.

Is any way possible for this?
Thank you.

**Comment by Reubenlee** [comment]
Hi,

i am using currently using elementor pro. i Just want to allow my client ( Author Role) to edit the text contents as well as uploading image to the page itself. However, i realise that client is also able to edit the layout of the individual widget in the page. Is there anyway, i can restrict edit of individual widget base on roles? 

For example, Allow editing of text box  but restrict user to change the anv menu layout. 

Thanks in advance 
Reuben

**Comment by black-eye** [comment]
+1

**Comment by portrayaloflife** [comment]
+1

**Comment by raulillana** [comment]
+1

**Comment by baitasite** [comment]
+1

**Comment by thomasmalley** [comment]
+ 1

**Comment by tristanleboss** [comment]
+1

**Comment by menathor** [comment]
+1

**Comment by Jcox1981** [comment]
**Request Feature** - allow for a more fine grained control within Role Manager please - similar to Visual Composer. 

+1

**Comment by webstarspro** [comment]
+1

**Comment by JamboMedia** [comment]
+1

**Comment by psahalot** [comment]
We built this plugin https://github.com/helloideabox/elementor-widget-manager it allows you to enable/disable Elementor widgets.

**Comment by sirucato** [comment]
> We built this plugin https://github.com/helloideabox/elementor-widget-manager it allows you to enable/disable Elementor widgets.

Hello @psahalot , sounds great. Is going to be available in de repository?
@black-eye , @jasonb4u, @Manuel-SG

**Comment by pingram3541** [comment]
Yes, this is a great idea.  As more and more features are added, we need a way to both choose what features we want to enable/disable via the back end and via roles, but also programmatically in which the theme or a helper plugin can control.  Not just visibility but code and dependencies too, the purpose is to restrict unwanted features but also remove unnecessary bloat.

**Comment by davidamarek** [comment]
+1

**Comment by rodolphebertozzo** [comment]
+¹

**Comment by dasigna** [comment]
+²

**Comment by benupham** [comment]
+1 Looked at the code base, both JS and PHP, and there is no way I can see to write a 3rd party plugin to provide more fine-grained control over what users can do in the Editor (that isn't very hacky). The key function being the JS userCan(). For example, we would like our client to be able to not just edit content but be able to add sections as Templates and rearrange sections -- but not design anything. Instead, we may have to build them custom Gutenberg blocks! Noooooo....

**Comment by paauserrat** [comment]
Hi, I'm currently using JetEngine Dynamic visability to show and hide elements to logged in and not logged in users.
![image](https://user-images.githubusercontent.com/73240206/132838154-81e3ba21-3d34-4ba0-98f6-55c8f3875b40.png)

The problem is that this plugin doesn't have an option to activate a fallback template to show when a user visits the site and the block isn't shown. It would be great to show a template when something is hidden. This will ensure the user knows something is hidden (for example, something like please deactivate your adblock to see this).

Some conditions:
![image](https://user-images.githubusercontent.com/73240206/132838417-1c1d4311-5460-4a75-a3a8-e1e8dc78bc14.png)
![image](https://user-images.githubusercontent.com/73240206/132838451-09d6695a-2320-4a20-8eff-e5fdb448f9b1.png)

**Comment by Suluuboi** [comment]
I found a way by just adding custom css to hide the buttons entirely.
```
.elementor .elementor-document-handle{
    display: none;
    visibility: hidden;
}
```

Or better yet if you would like to remove the edit buttons for all users but the admin you can install the plugin "Code Snippets" and add the code below.

```
add_action( 'wp_enqueue_scripts', 'hide_edit_buttons' );
function hide_edit_buttons(){
    if( !current_user_can('administrator') ){
        echo '<style>
            .elementor .elementor-document-handle{
                   display: none;
                   visibility: hidden;
             }
        </style>';
    }
}
```

**Comment by gervy28** [comment]
I made a slight revision to the above code in-order to hide the 'Back To WordPress Editor Button Only'


// Hide 'Back to WordPress Editor Button' for all roles, except admin 
add_action( 'admin_enqueue_scripts', 'sf_hide_back_to_wp_editor_buttons' );

if( ! function_exists( 'sf_hide_back_to_wp_editor_buttons' )) {
	function sf_hide_back_to_wp_editor_buttons(){
		if( !current_user_can('administrator') && is_admin() ){
			echo '<style>
				 #elementor-switch-mode-button {
				 	min-height: 0px !important;
					border: 0px !important;
				 }
				 
				 .elementor-switch-mode-on {
				 	display: none !important;
				 }
				 
			</style>';
		}
	}
}

**Comment by nicholaszein** [comment]
Hello!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.18.0, and improved with Elementor Pro 3.19.0! 🥳 

You can now use the Element Manager to decide what widgets should appear in the Editor and your website, and with Elementor Pro you can decide who can use those widgets. To learn more check the [documentation in the Help Center](https://elementor.com/help/element-manager/).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
